llvm/lib/Target/Primate/PrimateVLIWPacketizer.cpp

```diff
@@ -158,13 +220,49 @@ PrimatePacketizerList::addToPacket(MachineInstr &MI) {
 
 void PrimatePacketizerList::endPacket(MachineBasicBlock *MBB,
                                       MachineBasicBlock::iterator MI) {
+
+  // need to first generate the needed bypass ops
+  // generating bypass ops allows the fix up to x0 out the bypasses for free :>
+  MachineInstr* packet_breaking_instr = CurrentPacketMIs.back();
+  llvm::SmallVector<MachineInstr*, 6> generated_bypass_instrs;
+  MachineBasicBlock::iterator old_end_instr = MI;
+  bool push_branch_to_next_packet = false;
+  if(packet_breaking_instr->isBranch()) {
+    push_branch_to_next_packet = insertBypassOps(packet_breaking_instr, generated_bypass_instrs);
+  }
+
+  // if we push to the next packet then pop the branch from the back && set the isntr pointer back.
```

Comment:
Move block under above `isBranch()` check.

Reply:
Fixed in Fixed in [2c81b54](https://github.com/FAST-Research-Group/primate-arch-gen/pull/1/commits/2c81b54625f2a733519b99c357ec89f93d2fe3c5)